### PR TITLE
[RFC] Add entity update hooks

### PR DESCRIPTION
**Pull request by ranquild**
RFC https://www.notion.so/metabase/60-Add-hooks-for-updating-data-via-entity-actions-95311c27e8584780b745c8740b05ef9c

This is for `update`-compatible actions only. For creation we would need a separate base hook. Usage:

```
const { mutate: updateDatabase } = useDatabaseUpdate();

const handleSubmit = async (values: DatabaseData) => {
  await updateDatabase({ id }, values);
};
```

### CODE REVIEW

File: frontend/src/metabase/common/hooks/use-entity-update/use-entity-update.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  entityId?: TId;
+}
+
+export interface UseEntityUpdateProps<
+  TId,
+  TEntity,
+  TEntityInfo extends EntityInfo<TId>,
+  TEntityData,
+> {
+  update: (entityInfo: TEntityInfo, updates: Partial<TEntityData>) => Action;
```
Comment (ranquild) [inline frontend/src/metabase/common/hooks/use-entity-update/use-entity-update.ts:21]: Some update actions require more properties than only `id`, so it should be generic

**Review by kulyk — APPROVED** [review]
LGTM! Curious if you decided to abandon react-query-like API?

**Review by kulyk — APPROVED** [review]
LGTM! Curious if you decided to abandon react-query-like API?

**Comment by ranquild** [comment]
@kulyk it's very close. In react-query you'd get:
```
const { mutate: updateDatabase } = useDatabaseUpdateMutation();
updateDatabase({ id, ...values });
```

here:
```
const updateDatabase = useDatabaseUpdate();
updateDatabase({ id }, values);
```

**Comment by ranquild** [comment]
@kulyk hm, after thinking about it I think you're correct - in `react-query` there are additional fields like `isLoading` etc. Maybe we should not return the callback directly but also via `mutate` prop, I'll do that in a sec

**Pull request merged**